pyoozie/workflow.py

```diff
@@ -0,0 +1,43 @@
+# Copyright (c) 2017 "Shopify inc." All rights reserved.
+# Use of this source code is governed by a MIT-style license that can be found in the LICENSE file.
+import typing  # pylint: disable=unused-import
+
+from pyoozie import tags  # pylint: disable=unused-import
+
+
+class AbstractAction(object):
```

Comment:
> making AbstractAction an ABC,

Definitely

> adding an abstract method on_error() to it, and

👍 

> removing Kill from the hierarchy (since it doesn't make sense to have an error handler for Kill)?

Yeah `Kill` shouldn't be in the hierarchy, doesn't have error handling (it causes execution, parallel or not, to stop). The only concrete actions that we would support to start are `Shell`, `Email`, and `SubWorkflow` ([others can be found in various XML schemas](https://github.com/apache/oozie/blob/master/client/src/main/resources/oozie-workflow-0.5.xsd#L148)).